crates/provider/Cargo.toml

```diff
@@ -14,12 +14,13 @@ exclude.workspace = true
 [dependencies]
 alloy-json-rpc.workspace = true
 alloy-network.workspace = true
-alloy-rpc-client = { workspace = true, features = ["reqwest"] }
+alloy-rpc-client = { workspace = true, features = ["reqwest", "pubsub"] }
```

Comment:
> Need some help with fixing ci. Not sure why it's failing. `WsConnect` is to be imported from `alloy_transport_ws`. That's how it is on `main`
> 
> https://github.com/alloy-rs/alloy/blob/3108462e177741bbf2f72548fe1f84f98686b838/crates/rpc-client/tests/it/ws.rs#L4

`dep:alloy-transport-ws` is under the `ws` feature, but the test is `cfg(feature = "pubsub")` so when the pubsub feature is active, but the `ws` feature is not, there's an error. This is triggered by this feature line here, I believe

Reply:
basically, this feature activation causes the cargo resolver to activate ONLY `pubsub` on alloy-rpc-client, and not `ws`, which causes the alloy-transport-ws to be missing when the test runs. This feature should not be activated here, it should instead be activated under the `pubsub` feature n this crate

we should also change the cfg in `rpc-client/tests/it/main.rs`

Reply:
summary:
- remove this feature activation
- update cfgs in `rpc-client/tests/it/main.rs`

Reply:
cc @mattsse @DaniPopes should `ws` and `ipc` be default features for the `provider` crate?

Reply:
> summary:
> 
> * remove this `pubsub` feature activation
> * update cfgs in `rpc-client/tests/it/main.rs`

We need the `pubsub` feature enabled on alloy-rpc-client as it is required for `connect_pubsub`

Reply:
that implies that `connect_pubsub` should be cfg gated behind the alloy-provider `pubsub` feature

Reply:
https://github.com/alloy-rs/alloy/pull/342/files/072ff5ea130fe966660af428ff380a9febaac698#diff-2ba22fae76dc97fc89caff6fcd110d232b06d71b6d3c94717cdf80aa178cbe02R49

Reply:
and the WS/IPC components also need cfg gating

Reply:
> that implies that `connect_pubsub` should be cfg gated behind the alloy-provider `pubsub` feature

If I'm not mistaken, this means disabling the `connect_boxed` method as it uses `connect_pubsub` for ws and ipc. 

Not seeing a way to gate connect_pubsub using the pubsub feature without sacrificing `connect_boxed`.

Reply:
it means changing the behavior to produce ONLY enabled trandport types. eg. only http when pubsub is disabled